### PR TITLE
add leakage parameter #695

### PR DESCRIPTION
**Pull request by ThomaJu**
add 4 leakage parameter

### CODE REVIEW

File: ctapipe/image/leakage.py [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+            Number of photons in the second row of the border-pixel divided by all photons
+    '''
+    max_value = max(np.sum(geom.neighbor_matrix, axis=0))
+    leakage_border_pixel1 = 0
+    leakage_border_pixel2 = 0
+    leakage_border_photon1 = 0
+    leakage_border_photon2 = 0
+    size = 0
+    pixel_count = 0
+    for i in range(len(image)):
```
Comment (mackaiver) [inline ctapipe/image/leakage.py]: Maybe its faster if you only iterate over the non-zero pixel? I don't know. Shouldn't make a huge difference.

File: ctapipe/image/leakage.py [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+            leakage_border_pixel1 += 1
+            leakage_border_photon1 += image[i]
+        for j in range(len(geom.neighbor_matrix[i])):
+            if geom.neighbor_matrix[i][j] == 0:
+                continue
+            if np.sum(geom.neighbor_matrix[j]) != max_value:
+                leakage_border_pixel2 += 1
+                leakage_border_photon2 += image[i]
+                break
+    return leakage_border_pixel1 / pixel_count, leakage_border_pixel2 / pixel_count, leakage_border_photon1 / size, leakage_border_photon2 / size
```
Comment (mackaiver) [inline ctapipe/image/leakage.py]: @kosack Should these values go into a container or something? Like the hillas parameters?

File: ctapipe/image/leakage.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
+"""
+Leakage calculation
+"""
+
+import numpy as np
+__all__ = ['leakage']
+
+
+def leakage(geom, image):
+    r"""Calculating the leakage-values for a given image.
```
Comment (kosack) [inline ctapipe/image/leakage.py]: Can you define in the docstring what is the definition of a leakage parameter?

File: ctapipe/image/leakage.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+            pixel values
+
+    Returns
+    -------
+        leakage_border_pixel1:   float
+            Number of shower-pixel on the border divided by all shower-pixel
+        leakage_border_pixel2:   float
+            Number of shower-pixel in the second row of the border
+             divided by all shower-pixel
+        leakage_border_photon1:   float
```
Comment (kosack) [inline ctapipe/image/leakage.py]: I guess you mean photo-electrons and not photons here? It might be better to call it "intensity" as we do elsewhere.

**Comment by kosack** [comment]
Probably this could be sped up a lot of numpy operations rather than for-loops or even my just using numba.jit, if speed is a problem.

**Comment by kosack** [comment]
A somewhat more general question: for things like this, are we better off return a Container class with the names defined? In that case we can just add that container to the list of Containers to be written by a TableWriter, and automatically the results are added to the output file.  It is more complex though.

**Comment by kosack** [comment]
Please fix the docstring so that the docs build correctly (I think it's just a missing blank line):
```
Warning, treated as error:
/home/travis/build/cta-observatory/ctapipe/ctapipe/image/leakage.py:docstring of ctapipe.image.leakage:11:Block quote ends without a blank line; unexpected unindent.
Error: Error building sphinx doc
```

File: ctapipe/io/containers.py [hunk 5]
```diff
@@ -669,3 +670,17 @@ class HillasParametersContainer(Container):
...
     kurtosis = Field(nan, 'measure of the tailedness')
+
+
+class LeakageContainer(Container):
+    """
+    Leakage
+    """
+    leakage_pixel1 = Field(nan, ("shower-pixel on the border"
+                                 "devided by all shower-pixel"))
+    leakage_pixel1 = Field(nan, ("shower-pixel in the second row of the border"
```
Comment (mackaiver) [inline ctapipe/io/containers.py]: this should be leakage_pixel2

**Comment by kosack** [comment]
superceded by #783

**Pull request closed without merging**